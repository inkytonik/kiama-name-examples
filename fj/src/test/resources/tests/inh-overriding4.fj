class A extends Object {
  A() {
    super();
  }
  A mk(Object o1) {
    return new A();
  }
}
class B extends A {
  B() {
    super();
  }
  A mk(Object o1, Object o2) {
    return new B();
  }
}
new B().mk(new Object(), new Object())
