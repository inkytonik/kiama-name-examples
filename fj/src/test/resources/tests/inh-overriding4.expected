src/test/resources/tests/inh-overriding4.fj:5:15:info: o1 def: Object
  A mk(Object o1) {
              ^
src/test/resources/tests/inh-overriding4.fj:6:12:info: new A(): A
    return new A();
           ^
src/test/resources/tests/inh-overriding4.fj:13:3:error: method type (Object, Object) : A does not match overridden type (Object) : A
  A mk(Object o1, Object o2) {
  ^
src/test/resources/tests/inh-overriding4.fj:13:15:info: o1 def: Object
  A mk(Object o1, Object o2) {
              ^
src/test/resources/tests/inh-overriding4.fj:13:26:info: o2 def: Object
  A mk(Object o1, Object o2) {
                         ^
src/test/resources/tests/inh-overriding4.fj:14:12:info: new B(): B
    return new B();
           ^
src/test/resources/tests/inh-overriding4.fj:17:1:info: new B().mk ...: A
new B().mk(new Object(), new Object())
^
src/test/resources/tests/inh-overriding4.fj:17:1:info: new B(): B
new B().mk(new Object(), new Object())
^
src/test/resources/tests/inh-overriding4.fj:17:12:info: new Object ...: Object
new B().mk(new Object(), new Object())
           ^
src/test/resources/tests/inh-overriding4.fj:17:26:info: new Object ...: Object
new B().mk(new Object(), new Object())
                         ^
A
